auslib/db.py

```diff
@@ -1755,21 +1758,44 @@ def __init__(self, db, metadata, dialect):
         self.table.append_column(Column('data', BlobColumn(dataType), nullable=False))
         AUSTable.__init__(self, db, dialect, scheduled_changes=True, scheduled_changes_kwargs={"conditions": ["time"]})
 
-    def getPotentialRequiredSignoffs(self, affected_rows, transaction=None):
-        potential_required_signoffs = []
+    def getPotentialRequiredSignoffs(self, affected_rows, transaction=None, batch=False):
+        potential_required_signoffs = {}
+        rows = []
         for row in affected_rows:
             if not row:
                 continue
-            # Releases do not affect live updates on their own, only the
-            # product+channel combinations specified in Rules that point
-            # to them. We need to find these Rules, and then return _their_
-            # Required Signoffs.
-            info = self.getReleaseInfo(name=row["name"], transaction=transaction)
-            if info:
-                info = info[0]
-                for rule_id in info["rule_ids"]:
-                    rule = self.db.rules.select(where=[self.db.rules.rule_id == rule_id], transaction=transaction)[0]
-                    potential_required_signoffs.extend(self.db.rules.getPotentialRequiredSignoffs([rule], transaction=transaction))
+            rows.append(row)
+        info = self.getReleaseInfo(name=[row['name'] for row in rows] if batch else rows[0]['name'], transaction=transaction, batch=batch)
+        # Releases do not affect live updates on their own, only the
+        # product+channel combinations specified in Rules that point
+        # to them. We need to find these Rules, and then return _their_
+        # Required Signoffs.
+        if info:
+            # get all rules as a union
+            q_rules = [[self.db.rules.rule_id == rule_id] for row in info for rule_id in row['rule_ids']]
+            all_rules = self.db.rules.select(where=q_rules, transaction=transaction)
```

Comment:
Good idea here - it looks like it reduces the number of times we call `Rules.getPotentialRequiredSignoffs`.

Reply:
Thanks 👍